flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StateTransitions.java

```diff
@@ -164,6 +166,6 @@ CompletableFuture<String> goToStopWithSavepoint(
     interface ToWaitingForResources extends StateTransitions {
 
         /** Transitions into the {@link WaitingForResources} state. */
-        void goToWaitingForResources();
+        void goToWaitingForResources(@Nullable ExecutionGraph executionGraph);
```

Comment:
nit: (this appears to affect all implementations)

```suggestion
        void goToWaitingForResources(@Nullable ExecutionGraph previousExecutionGraph);
```